pkg/orchestration/wiring/wiringutil/compute/secret.go

```diff
@@ -0,0 +1,110 @@
+package compute
+
+import (
+	"crypto"
+	"encoding/hex"
+	"io"
+	"strings"
+
+	smith_v1 "github.com/atlassian/smith/pkg/apis/smith/v1"
+	"github.com/atlassian/voyager"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil"
+	"github.com/atlassian/voyager/pkg/orchestration/wiring/wiringutil/knownshapes"
+	"github.com/pkg/errors"
+)
+
+type BindingResult struct {
+	ResourceName            voyager.ResourceName
+	BindableEnvVarShape     *knownshapes.BindableEnvironmentVariables
+	CreatedBindingFromShape smith_v1.Resource
+}
+
+func GenerateEnvVars(renameEnvVar map[string]string, bindingResults []BindingResult) ([]smith_v1.Reference, map[string]string, error) {
+	originalEnvVars := map[string]string{}
+	dependencyReferences := []smith_v1.Reference{}
+
+	for _, bindingResult := range bindingResults {
+		prefix := bindingResult.BindableEnvVarShape.Data.Prefix
+		bindingName := bindingResult.CreatedBindingFromShape.Name
+		resourceName := bindingResult.ResourceName
+
+		for envVarKey, path := range bindingResult.BindableEnvVarShape.Data.Vars {
```

Comment:
Does it mean that list of all variables must be known in advance even before getting response for Service Binding request?

I'm fine with that. I don't like how we encode that list. I'll use Micros2 | Autowiring2 room to discuss this.

Reply:
We can discuss this separately even after this PR gets merged.

Reply:
Discussed in room, resolving conversation